# Installing Transformers

In [16]:
# Installing the neccesary packages for implementing the model
!pip install transformers

# Model Importing

In [17]:
from transformers import pipeline

### Initializing Model

In [18]:
summarizer = pipeline("summarization", model="facebook/bart-large-cnn")


We are taking summarization from the pipeline out of question-answering, image-classification and so on. and we are we are using the model as bart-large-cnn which has been created by facebook and tuned by CNN


# Model testing

### Article 1

In [19]:
ARTICLE_1 = '''The US women's national team stood as a group during the anthem prior to their SheBelieves Cup game on February 21 after some knelt in the tournament opener on February 18.
Team members said they were past the protesting phase of the anthem debate but are still committed to fighting to end systemic racism.
National Football League player Colin Kaepernick began kneeling during the anthem in 2016 to protest racial injustice. Other players joined him until team owners banned the practice. That policy was reversed in 2020 during a wave of national protests over racism and police brutality after the death of George Floyd in May.
In June, US Soccer's board of directors voted to repeal the no-kneeling policy, which was put in place after Megan Rapinoe knelt in 2016.
That board vote required confirmation by the wider US Soccer governing body, which it received on Saturday.'''

In [20]:
article1 = summarizer(ARTICLE_1, max_length=130, min_length=30)

print(article1)


[{'summary_text': "The US women's national team stood as a group during the anthem prior to their SheBelieves Cup game on February 21. Team members said they were past the protesting phase of the anthem debate but are still committed to fighting to end systemic racism."}]


We are calling the summarizer we initialized before to run in in this article with parameter min and max length.

1.   min_length = the least number of words should be in a article.
2.   max_length = the maximum number of words to be in a article.

In [21]:
len(ARTICLE_1)

877

In [22]:
ls = str(article1)
ls = ls.split()
len(ls)

45

From 877 words to we are making it to 45

### Article 2

In [23]:
ARTICLE_2 = '''Many of the police departments involved in these efforts are working with charity organizations and former members of the US military. Some sources with direct knowledge of the varied efforts -- but who are not involved -- spoke to CNN on the condition of anonymity due to concerns about potential legal questions the effort could raise.
It's unprecedented, experts say, for US law enforcement agencies to donate police protective equipment and military-grade gear to a foreign country involved in an ongoing war. The effort also raises questions about the roles of police departments and whether, as domestic law enforcement agencies, they should send equipment to a foreign conflict outside of their jurisdiction.
Because there's no central coordinating group, there's not an easy way to say what's being shipped or whether it's subject to export regulations.
The Ukrainian American Coordinating Council (UACC), a non-profit organization, is at the center of one effort to send regulated, military-grade and police protective items to Ukraine, including ballistic helmets, hard plates, soft armor inserts and vests, the group says. 
When Russia's invasion of Ukraine began, the UACC moved quickly to obtain a license from the Department of Commerce to export level III ballistic vests and helmets, as well as a special authorization from the State Department for level IV vests in a limited capacity, according to the organization.
Level III armor is the first level of body armor that provides protection against rifle rounds and level IV is rated by the National Institute of Justice as the highest level of ballistic protection.'''

In [24]:
article2 = summarizer(ARTICLE_2, max_length=150, min_length=50)
print(article2)


[{'summary_text': "Experts say it's unprecedented for US law enforcement agencies to donate police protective equipment to a foreign country. Many of the police departments involved in these efforts are working with charity organizations and former members of the US military. Because there's no central coordinating group, it's not easy to say what's being shipped or whether it's subject to export regulations."}]


In [25]:
ls = str(article2)
ls = ls.split()
print(len(ls))
print(len(ARTICLE_2))

61
1633


From 1633 words to we are making it to 61

### Article 3

In [26]:
ARTICLE_3 ='''Hugging Face has raised a $15 million funding round led by Lux Capital. The company first built a mobile app that let you chat with an artificial BFF, a sort of chatbot for bored teenagers. More recently, the startup released an open-source library for natural language processing applications. And that library has been massively successful.
A.Capital, Betaworks, Richard Socher, Greg Brockman, Kevin Durant and others are also participating in today’s funding round.
Hugging Face launched its original chatbot app back in early 2017. After months of work, the startup wanted to prove that chatbots don’t have to be a glorified command line interface for customer support.
With the app, you could generate a digital friend and text back and forth with your companion. And it wasn’t just about understanding what you meant — the app tried to detect your emotions to adapt answers based on your feelings.
It turns out that the technology behind that chatbot app is solid. As Brandon Reeves from Lux Capital wrote, there’s been a ton of progress when it comes to computer vision and image processing, but natural language processing has been lagging behind.
Hugging Face’s open-source framework Transformers has been downloaded over a million times. The GitHub project has amassed 19,000 stars, proving that the open-source community thinks this is a useful brick to build upon. Researchers at Google, Microsoft and Facebook have been playing around with it.
Some companies even use it in production, such as challenger bank Monzo for its customer support chatbot and Microsoft Bing. You can leverage Transformers for text classification, information extraction, summarization, text generation and conversational artificial intelligence.
With today’s funding round, the company plans to triple its headcount in New York and Paris'''

In [27]:
article3 = (summarizer(ARTICLE_3, max_length=100, min_length=25))
print(article3)

[{'summary_text': 'Hugging Face has raised a $15 million funding round led by Lux Capital. The company first built a mobile app that let you chat with an artificial BFF. More recently, the startup released an open-source library for natural language processing applications.'}]


In [28]:
ls = str(article3)
ls = ls.split()
print(len(ls))
print(len(ARTICLE_2))

42
1633


From 1633 words to we are making it to 42